eng/pipelines/agent-pool-migration.yml

```diff
@@ -0,0 +1,138 @@
+# The pipeline is mainly used to migrate our agent pool from one version to another version. 
```

Comment:
I see the pipeline is called [`automation - migrate-new-pool`](https://dev.azure.com/azure-sdk/internal/_build?definitionId=5791&_a=summary).

How about we rename it to `automation - agent-pool-migration`, and thus make it easier to figure out the pipeline based on this file name?

Reply:
Done. I thought I changed the name. Might not save after renaming.